website/docs/plugin/framework/functions/implementation.mdx

```diff
@@ -207,7 +207,9 @@ func (f *ExampleFunction) Definition(ctx context.Context, req function.Definitio
         Parameters: []function.Parameter{
             function.BoolParameter{},
         },
-        VariadicParameter: function.StringParameter{},
+        VariadicParameter: function.StringParameter{
+			Name: "variadic_param",
+        },
```

Comment:
This was something interesting I found, Terraform core validates that the variadic parameter name is different from any of the normal parameters.

Should we change the default `Name` for variadic parameter to avoid this conflict? Also should we add validation for this, maybe when we introduce the user-supplied validation?

```bash
│ Error while loading schemas for plugin components: Failed to obtain provider schema: Could not load the schema for provider registry.terraform.io/austinvalle/sandbox:
│ provider registry.terraform.io/austinvalle/sandbox function "dynamic_test" reuses name "param" for both parameter 0 and its variadic parameter..
```

Reply:
Oh, very interesting! I would not have necessarily expected that. Let's follow up with that team to see if there is a technical reason for that or if it is more for aesthetics/readability of signatures.

Reply:
I opened https://github.com/hashicorp/terraform-plugin-framework/issues/926 to consider addressing this 👍🏻